readers/mongodb/messages.go

```diff
@@ -101,7 +101,10 @@ func fmtCondition(chanID string, rpm readers.PageMetadata) bson.D {
 	if err != nil {
 		return filter
 	}
-	json.Unmarshal(meta, &query)
+	err = json.Unmarshal(meta, &query)
+	if err != nil {
+		return filter
+	}
```

Comment:
Inline `if err := json.Unmarshal(....`. What if err is nil?

Reply:
I inlined the err.
And if the err is nil, the query is ranged over in a for loop.